### PR TITLE
Add PyTorch 2.0 to ORT transformer benchmarking

### PR DESCRIPTION
**Pull request by kunal-vaishnavi**
### Description
This PR adds PyTorch 2.0 as an option when running the ORT transformer benchmarking script.

### Motivation and Context
PyTorch released [PyTorch 2.0](https://pytorch.org/get-started/pytorch-2.0/) in the nightly binaries and a stable release of PyTorch 2.0 is expected in March 2023.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/benchmark.py [hunk 1]
```diff
@@ -366,13 +367,13 @@ def run_pytorch(
                     device=device,
                 )
                 try:
-                    inference = torch.jit.trace(model, input_ids) if torchscript else model
+                    inference = torch.jit.trace(model, input_ids) if torchscript else torch.compile(model) if torch2 else model
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/benchmark.py:370]: Does stable version of pytorch has torch.compile API? You might check whether this script could run in pytorch 1.*.

Comment (kunal-vaishnavi) [reply onnxruntime/python/tools/transformers/benchmark.py:370]: Added a check for PyTorch version

File: onnxruntime/python/tools/transformers/benchmark.py [hunk 2]
```diff
@@ -779,6 +779,10 @@ def main():
     enable_onnxruntime = "onnxruntime" in args.engines
     enable_tensorflow = "tensorflow" in args.engines
 
+    if enable_torch2 and torch.__version__ < "2.0.0":
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/benchmark.py:782]: use packaging.version to compare version like `version.parse(torch.__version__) < version.parse("2.0.0")`

Comment (kunal-vaishnavi) [reply onnxruntime/python/tools/transformers/benchmark.py]: Fixed

**Review by tianleiwu — DISMISSED** [review]

**Review by hanbitmyths — DISMISSED** [review]

**Review by hanbitmyths — APPROVED** [review]

**Pull request merged**